lib/impl/dataSources/updateCache.js

```diff
@@ -21,7 +21,11 @@ function updateValidAndInvalidDataSources(dataSourcesToSplit, currentInvalidData
   logger.debug("dataSourcesToSplit", dataSourcesToSplit);
   currentInvalidDataSources = currentInvalidDataSources || [];
   var validInvalid = _.partition(dataSourcesToSplit, function(dataSourceUpdateData) {
-    return !dataSourceUpdateData.error;
```

Comment:
My suggestion
if ( dataSourceUpdateData && dataSourceUpdateData.error ){
return dataSourceUpdateData.error;
}

PS. In JS/Node to check if the variable is != of undefined you can just if (var)

Reply:
The original exceptions being raised are the following:
```
TypeError: Cannot read property 'error' of undefined
```
There seem to be situations where dataSourceUpdateData is being passed and is undefined, and calling .error on it raises an exception. So the suggested fix you make will not resolve this see the following as an example:

```
> var f = function(test){ test.a; }
undefined
> f
[Function]
> f(undefined);
TypeError: Cannot read property 'a' of undefined
    at f (repl:1:29)
    at repl:1:1
    at REPLServer.defaultEval (repl.js:262:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:431:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:211:10)
    at REPLServer.Interface._line (readline.js:550:8)
> 
```

Now compare to the fix I'm proposing:

```
> var f2 = function(test) {
... if(typeof test === 'undefined'){
..... return true;
..... }
... else{
..... return test.a;
..... }
... }
undefined
> 
> 
> f2({a:1});
1
> f2(undefined);
true
> 
```

Reply:
Opps my bad! I'm wrong here, this does appear to work! 

```
> var f = function(test){
... if(test){
..... return !test.a;
..... }
... return true;
... }
undefined
> f(undefined);
true
> f({a:false});
true
> f({a:true});
false
> 
```